packages/core/js-client/src/api.ts

```diff
@@ -15,182 +15,158 @@
  */
 
 import type {
-  FnConfig,
   FunctionCallDef,
+  JSONValue,
+  SimpleTypes,
+  ArrowWithoutCallbacks,
   ServiceDef,
-  PassedArgs,
-  ServiceImpl,
 } from "@fluencelabs/interfaces";
-import { getArgumentTypes } from "@fluencelabs/interfaces";
 
+import { CallAquaFunctionConfig } from "./compilerSupport/callFunction.js";
+import {
+  aqua2ts,
+  ts2aqua,
+  wrapFunction,
+  wrapServiceFunction,
+} from "./compilerSupport/conversions.js";
+import { ServiceImpl } from "./compilerSupport/types.js";
 import { FluencePeer } from "./jsPeer/FluencePeer.js";
 
 import { callAquaFunction, Fluence, registerService } from "./index.js";
 
-export const isFluencePeer = (
-  fluencePeerCandidate: unknown,
-): fluencePeerCandidate is FluencePeer => {
-  return fluencePeerCandidate instanceof FluencePeer;
-};
-
 /**
  * Convenience function to support Aqua `func` generation backend
  * The compiler only need to generate a call the function and provide the corresponding definitions and the air script
  *
- * @param rawFnArgs - raw arguments passed by user to the generated function
+ * @param args - raw arguments passed by user to the generated function
  * @param def - function definition generated by the Aqua compiler
  * @param script - air script with function execution logic generated by the Aqua compiler
  */
 export const v5_callFunction = async (
-  rawFnArgs: unknown[],
+  args: (JSONValue | ((...args: JSONValue[]) => JSONValue))[],
```

Comment:
wdyt about declaring `((...args: JSONValue[]) => JSONValue)` as a separate type alias? Right now it's hard to parse meaning through all these parentheses

Reply:
fixed